src/cli/commands/lint.ts

```diff
@@ -105,16 +105,24 @@ async function lint(name: string, flags: any, command: Lint, customRules?: RuleC
   const spec: IParserResult = await readParsable(name, flags.encoding);
 
   const spectral = new Spectral();
+  command.log('Setting up functions...');
+  if (parseInt(spec.data.swagger) === 2) {
+    command.log('Adding OpenAPI 2.0 (Swagger) functions');
+    spectral.addFunctions(oas2Functions());
+  } else if (parseInt(spec.data.openapi) === 3) {
+    command.log('Adding OpenAPI 3.x functions');
+    spectral.addFunctions(oas3Functions());
+  }
+
```

Comment:
Because you can't import functions using the `yaml` file. Imagine someone imported the oas3 ruleset using the `--ruleset` param. In such case we disabled the automatic detection and wouldn't import oas3 functions (w/o functions that file would fail to execute).

This fixes that issue (we always detect functions automatically but only detect rules if a ruleset file was not provided). We will be able to remove this code when we extend the `yaml` files with `functions` functionality.